include/mbgl/util/thread.hpp

```diff
@@ -127,13 +140,17 @@ class Thread : public Scheduler {
 private:
     MBGL_STORE_THREAD(tid);
 
-    void schedule(std::weak_ptr<Mailbox> mailbox) override {
-        loop->schedule(mailbox);
+    void schedule(std::weak_ptr<Mailbox> mailbox_) override {
+        assert(loop);
+        loop->schedule(mailbox_);
     }
+    
+    std::unique_ptr<Actor<Object>> object;
 
     std::thread thread;
-    std::unique_ptr<Actor<Object>> object;
 
+    std::future<void> running;
```

Comment:
Make this `std::future<RunLoop*>` and replace the uses of the `loop` member with `running.get()`.

Reply:
The reason I didn't do it this way is that from http://en.cppreference.com/w/cpp/thread/future/get, it seems like `std::future::get()` is destructive, so we couldn't just treat `running.get()` as an accessor.  Given that, this approach seemed just as simple, but maybe there's still a cleaner way